### PR TITLE
[Chore] - Exam timer tests

### PR DESCRIPTION
**Pull request by FuchsDominik**
### Checklist
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)

### Motivation and Context

Client test coverage improvements

### Description

- Added exam-timer client tests
- After consultation with the developer of this component, changed `duration.milliseconds()` to `duration.asMilliseconds()` as it turns out to be less error-prone and serves the same logic (For more infos, check https://momentjs.com/docs/#/durations/milliseconds/)

### Test Coverage

exam-timer.component.ts: 82%

### CODE REVIEW

File: src/test/javascript/spec/component/exam/exam-timer.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        fixture = TestBed.createComponent(ExamTimerComponent);
+        component = fixture.componentInstance;
+        dateService = TestBed.inject(ArtemisServerDateService);
+        component.endDate = inFuture;
+    });
+
+    it('should call ngOnInit', () => {
+        spyOn(dateService, 'now').and.returnValue(now);
+        component.criticalTime = moment.duration(200);
+        expect(component.endDate).to.equal(inFuture);
```
Comment (stefanwaldhauser) [inline src/test/javascript/spec/component/exam/exam-timer.component.spec.ts:49]: Seems unncessary as you are setting it this way in the beforeEach

File: src/test/javascript/spec/component/exam/exam-timer.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        component.ngOnInit();
+        expect(component).to.be.ok;
+        expect(component.isCriticalTime).to.be.true;
+    });
+
+    it('should update display times', () => {
```
Comment (stefanwaldhauser) [inline src/test/javascript/spec/component/exam/exam-timer.component.spec.ts:55]: I added a test in 486fa378fe75ae455478b8f1a15d009470622a35 that also tests the async behaviour.

**Review by stefanwaldhauser — CHANGES_REQUESTED** [review]

File: src/test/javascript/spec/component/exam/exam-timer.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    let dateService: ArtemisServerDateService;
+
+    const now = moment();
+    const inFuture = moment().add(100, 'ms');
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule],
+            declarations: [ExamTimerComponent, MockPipe(TranslatePipe)],
+            providers: [
```
[the remark this answers is not in the document]
Comment (FuchsDominik) [reply src/test/javascript/spec/component/exam/exam-timer.component.spec.ts:32]: Very true, forgot to check that, thanks

File: src/test/javascript/spec/component/exam/exam-timer.component.spec.ts (hunk 1 above)
Comment (FuchsDominik) [reply src/test/javascript/spec/component/exam/exam-timer.component.spec.ts:49]: Yeah, thought it doesn't hurt as i had some weird behaviour when started testing but sure, not necessarily needed ;)

File: src/test/javascript/spec/component/exam/exam-timer.component.spec.ts (hunk 2 above)
Comment (FuchsDominik) [reply src/test/javascript/spec/component/exam/exam-timer.component.spec.ts:55]: Sure, test looks good :) And also covers an aspect i did not quite catch in my tests

**Review by stefanwaldhauser — APPROVED** [review]
Aproooooooove

**Review by FrankeLukas — APPROVED** [review]

**Review by krusche — APPROVED** [review]

**Pull request merged**